cmd/kube-scheduler/app/options/options.go

```diff
@@ -138,6 +138,9 @@ func splitHostIntPort(s string) (string, int, error) {
 func newDefaultComponentConfig() (*kubeschedulerconfig.KubeSchedulerConfiguration, error) {
 	versionedCfg := kubeschedulerconfigv1beta1.KubeSchedulerConfiguration{}
 	versionedCfg.DebuggingConfiguration = *configv1alpha1.NewRecommendedDebuggingConfiguration()
+	// Use lease-based leader election to reduce cost.
+	// The default endpoints-leases one has already been used for couple releases.
+	versionedCfg.LeaderElection.ResourceLock = resourcelock.LeasesResourceLock
```

Comment:
be specific about which release we switched the default to endpointslease. also, why not change the default in SetDefaults_KubeSchedulerConfiguration instead of here?